doc/decisions/cryptograhic_key_handling.md

```diff
@@ -0,0 +1,65 @@
+# Cryptographic Key Handling
+
+## Issue
+
+The crypto plugin applies cryptographic operations to Keys and KeySets.
+In order to do that it needs keys and initialization vectors (IV).
+The problem is how to retrieve or derivate those keys in a safe way and how 
+to pass them on to the underlying crypto libraries (OpenSSL and libgcrypt 
+at the time of writing).
+
+## Constraints
+
+The solution must be reasonable to the user (not just experimental setups).
+
+The key handling should follow best practises when dealing with cryptographic
+operations, thus:
+
+- the combination of key and IV must not be used twice for symmetric encryption
+- the key and IV material must be as random as possible (using an SNRG)
+- the key is not exposed to other Elektra modules or the user
+
+## Assumptions
+
+- The user knows or is willing to learn how to use GPG.
+- The user is willing to configure `gpg` and `pinentry` on her system, as it does not always run out of the box (depending on the distribution)
+
+## Considered Alternatives
+
+We considered passing the key and IV in form of plugin configuration and meta-keys, but this approach violates our constraints.
```

Comment:
Give some more alternatives, e.g. having password as plugin config, directly using gpg-agent or pcscd,...

Also argument better: which specific constraint is violated.